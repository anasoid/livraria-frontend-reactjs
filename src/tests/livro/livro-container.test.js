import React from 'react';
import renderer from 'react-test-renderer';
import {mount} from 'enzyme';
import { Provider } from "react-redux";
import configureStore from 'redux-mock-store';
import thunk from 'redux-thunk';
import LivroContainer from '../../components/livro/livro-container';
import LivroTable from '../../components/livro/livro-table';
import LivroForm from '../../components/livro/livro-form';

import initialState from '../util/initial-state';
const middlewares = [thunk];
const mockStore = configureStore(middlewares);
const store = mockStore(initialState);
const component = (
  <Provider store={store}>
    <LivroContainer/>
  </Provider>
);

describe('test LivroConteiner component', () => {

  let wrapper;

  beforeEach(() => {
    wrapper = mount(component);
  });

  it('should renders without crashing ', () => {
    const tree = renderer.create(component)
      .toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('should be "Novo Livro" h1 title ', () => {
    expect(wrapper.find('h1').first()).toHaveText('Novo Livro');
  });

  it('should be render LivroForm', () => {
    expect(wrapper.find(LivroForm)).toExist();
  });

  it('should be render LivroTable', () => {
    expect(wrapper.find(LivroTable)).toExist();
  });

});