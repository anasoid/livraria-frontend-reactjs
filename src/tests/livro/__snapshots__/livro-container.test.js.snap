// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`test LivroConteiner component should renders without crashing  1`] = `
<div>
  <h1>
    Novo Livro
  </h1>
  <div>
    <form
      onSubmit={[Function]}
    >
      <label
        htmlFor="livroTitulo"
      >
        Titulo: 
      </label>
      <input
        id="livroTitulo"
        name="titulo"
        onChange={[Function]}
      />
      <br />
      <label
        htmlFor="livroIsbn"
      >
        ISBN: 
      </label>
      <input
        id="livroIsbn"
        name="isbn"
        onChange={[Function]}
      />
      <br />
      <label
        htmlFor="livroPreco"
      >
        Preço: 
      </label>
      <input
        id="livroPreco"
        name="preco"
        onChange={[Function]}
      />
      <br />
      <label
        htmlFor="livroDataLancamento"
      >
        Data de Lançamento: 
      </label>
      <input
        id="livroDataLancamento"
        name="dataLancamento"
        onChange={[Function]}
      />
      <br />
      <label
        htmlFor="livroAutores"
      >
        Selecione o Autor: 
      </label>
      <select
        id="livroAutores"
        name="selectAutor"
        onChange={[Function]}
        value=""
      >
        <option
          value=""
        >
          Selecione
        </option>
      </select>
      <button
        id="btnAddAutor"
        onClick={[Function]}
        type="button"
      >
        Adicionar Autor
      </button>
      <br />
      <label>
        Autores: 
      </label>
      <ul />
      <br />
      <button
        disabled={true}
        id="btnGravar"
        type="submit"
      >
        Gravar Livro
      </button>
    </form>
  </div>
  <br />
  <div>
    <table>
      <caption>
        Livros
      </caption>
      <thead>
        <tr>
          <th>
            Titulo
          </th>
          <th>
            ISBN
          </th>
          <th>
            Preço
          </th>
          <th>
            Data Lançamento
          </th>
        </tr>
      </thead>
      <tbody />
    </table>
  </div>
</div>
`;
